R/pruneScores.R

```diff
@@ -54,25 +53,31 @@
 #' # Running the SingleR() example.
 #' example(SingleR, echo=FALSE)
 #'
-#' summary(pruneScores(pred$scores))
+#' summary(pruneScores(pred))
 #'
 #' # Less stringent:
-#' summary(pruneScores(pred$scores, min.diff.med=0))
-#' summary(pruneScores(pred$scores, nmads=5))
+#' summary(pruneScores(pred, min.diff.med=0))
+#' summary(pruneScores(pred, nmads=5))
+#' 
+#' # More stringent:
+#' summary(pruneScores(pred, min.diff.med=0.1))
+#' summary(pruneScores(pred, nmads=2))
+#' summary(pruneScores(pred, min.diff.next=0.1))
 #' 
 #' @export
 #' @importFrom DelayedMatrixStats rowMedians 
 #' @importFrom DelayedArray DelayedArray rowMaxs
 #' @importFrom stats median mad
-pruneScores <- function(scores, min.diff.med = 0.05, nmads=3, min.diff.next = 0) {
-    if (is(scores, "DataFrame")){
-      scores <- scores$scores
-    }
+pruneScores <- function(results, min.diff.med = 0.05, nmads=3, min.diff.next = 0) {
+    if (is(results, "DataFrame")){
+      scores <- results$scores
+      tune.scores <- results$tuning.scores
+    } else { stop("full results output of SingleR/classifySingleR is required")}
```

Comment:
No need for that check, just crash out if it's not expected. It would actually take _more_ effort to use `$scores`, so I don't think we need protection against that case (any more than some other weird thing that a user might put in).

Reply:
if check and else{stop(...)} removed.